actions/is_dpkg_active

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python3
+import fcntl
+
+def is_dpkg_active():
+    """
+    Check whether ``apt-get`` or ``dpkg`` is currently active.
+
+    This works by checking whether the lock file ``/var/lib/dpkg/lock`` is
+    locked by an ``apt-get`` or ``dpkg`` process, which in turn is done by
+    momentarily trying to acquire the lock. This means that the current process
+    needs to have sufficient privileges.
+
+    :returns: ``True`` when the lock is already taken (``apt-get`` or ``dpkg``
+              is running), ``False`` otherwise.
+    :raises: :py:exc:`exceptions.IOError` if the required privileges are not
+             available.
+
+    .. note:: ``apt-get`` doesn't acquire this lock until it needs it, for
+              example an ``apt-get update`` run consists of two phases (first
+              fetching updated package lists and then updating the local
+              package index) and only the second phase claims the lock (because
+              the second phase writes the local package index which is also
+              read from and written to by ``dpkg``).
+    """
+    with open('/var/lib/dpkg/lock', 'w') as handle:
+        try:
+            fcntl.lockf(handle, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
Thanks a lot for your extensive explanation. It's a more thorny issue than I had hoped :). I've deleted the function, I simply wasn't aware that we had another implementation. I've moved the sudo-calling `is_package_manager_busy()` into `actions/packages`.
Maybe noteworthy: While we now re-use that action, there's duplicated code of the "secondary" function `is_package_manager_busy()` in `plinth/views.py`, that I essentially copied from `plinth/modules/upgrades/views.py` ... maybe that should be unified, too?